### PR TITLE
Squash timeline

### PR DESCRIPTION
**Pull request by tonyhb**
Squashes the function log timeline to hide noisy events, eg. "scheduled", "started" if the step has finished.

### CODE REVIEW

File: pkg/devserver/devserver.go [hunk 1]
```diff
@@ -45,12 +46,15 @@ func New(ctx context.Context, opts StartOpts) error {
...
 	// service starting.
-	el, err := prepareDockerImages(ctx, opts.RootDir)
-	if err != nil {
-		return err
+	el, _ := inmemorydatastore.NewFSLoader(ctx, opts.RootDir)
+	if opts.Docker {
+		var err error
+		if el, err = prepareDockerImages(ctx, opts.RootDir); err != nil {
+			return err
+		}
```
Comment (jpwilliams) [inline pkg/devserver/devserver.go:54]: 🙌 Awesome!

File: pkg/coreapi/graph/resolvers/events.go [hunk 2]
```diff
@@ -20,7 +21,14 @@ func (r *queryResolver) Event(ctx context.Context, query models.EventQuery) (*mo
 	}
 
 	evt := evts[0]
+
 	createdAt := time.UnixMilli(evt.Timestamp)
+	if evt.Timestamp == 0 {
```
Comment (jpwilliams) [inline pkg/coreapi/graph/resolvers/events.go:26]: Is it still a bug that `evt.Timestamp` isn't defined here?

File: cmd/commands/dev.go [hunk 3]
```diff
@@ -22,6 +22,7 @@ func NewCmdDev() *cobra.Command {
 	cmd.Flags().StringP("port", "p", "8288", "port to run the API on")
 	cmd.Flags().String("dir", ".", "directory to load functions from")
 	cmd.Flags().StringSliceP("sdk-url", "u", []string{}, "SDK URLs to load functions from")
+	cmd.Flags().Bool("docker", false, "enable docker-based steps")
```
Comment (jpwilliams) [inline cmd/commands/dev.go:25]: Do we need to alter any docs for this?

File: pkg/execution/executor/service.go [hunk 4]
```diff
@@ -358,7 +358,7 @@ func (s *svc) handleQueueItem(ctx context.Context, item queue.Item) error {
 		// and this is a no-op - things should be atomic where possible.
 		//
 		// TODO: Add a unit test to ensure WG is 0 at the end of execution.
-		if err := s.state.Scheduled(ctx, item.Identifier, next.Incoming, 0, nil); err != nil {
+		if err := s.state.Scheduled(ctx, item.Identifier, next.Incoming, 0, &at); err != nil {
```
Comment (jpwilliams) [inline pkg/execution/executor/service.go:361]: Ah - good catch

File: ui/src/components/Event/Stream.tsx [hunk 5]
```diff
@@ -3,6 +3,7 @@ import { selectEvent } from "../../store/global";
 import { useAppDispatch, useAppSelector } from "../../store/hooks";
 import TimelineFeedContent from "../Timeline/TimelineFeedContent";
 import TimelineRow from "../Timeline/TimelineRow";
+import { decodeTime } from 'ulid'
```
Comment (jpwilliams) [inline ui/src/components/Event/Stream.tsx:6]: Dangling import?

File: ui/src/components/Function/Stream.tsx [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import {
+    FunctionRunEvent,
```
Comment (jpwilliams) [inline ui/src/components/Function/Stream.tsx:2]: Dangling import?

File: ui/src/components/Time.tsx [hunk 7]
```diff
@@ -1,23 +1,27 @@
...
 
-  return <TimeAgo date={date} formatter={formatter} title={tooltip} />;
+  // Refresh each second such that this component re-renders times appropriately.
+  const [time, setTime] = useState(new Date().valueOf());
+  useEffect(() => {
+    const interval = setInterval(() => { setTime(new Date().valueOf()) }, 1000);
+    return () => { clearInterval(interval); };
+  }, []);
+
+  return <TimeAgo date={date} formatter={formatter} title={tooltip} key={time} />;
```
Comment (jpwilliams) [inline ui/src/components/Time.tsx:26]: [react-timeago](https://www.npmjs.com/package/react-timeago) should already do this

**Review by jpwilliams — COMMENTED** [review]
Awesome! We have [INN-768](https://linear.app/inngest/issue/INN-768); this feels like a sensible mid point to make the timelines easy to view. 🙌

Only a couple of notes/nits

File: pkg/coreapi/graph/resolvers/events.go (hunk 2 above)
Comment (tonyhb) [reply pkg/coreapi/graph/resolvers/events.go:26]: It is - that needs to be fixed with additional unit tests in a followup PR

File: cmd/commands/dev.go (hunk 3 above)
Comment (tonyhb) [reply cmd/commands/dev.go:25]: This is legacy and isn't supported for `inngest dev`, so we shouldn't!

File: ui/src/components/Event/Stream.tsx (hunk 5 above)
Comment (tonyhb) [reply ui/src/components/Event/Stream.tsx:6]: thanks! :derp:

File: ui/src/components/Time.tsx (hunk 7 above)
Comment (tonyhb) [reply ui/src/components/Time.tsx:26]: It doesn't, for some reason :/

File: pkg/devserver/devserver.go (hunk 1 above)
Comment (djfarrelly) [reply pkg/devserver/devserver.go:54]: Big step here! 🔥

**Review by djfarrelly — APPROVED** [review]

**Pull request merged**